### ISSUE TITLE
No data populated on Debian11 with lightdm and xfce4 wm

### ISSUE THREAD

**Issue by gwhizoftv** [DavidoTek/ProtonUp-Qt#130]
**Describe the bug**  
WHen double-clicking Flatpak ref, we get an error "Failed to Install/ File not supported"
When Running the AppImage, we get empty dialogues

**To Reproduce**  
FlatPak
1. On Debian 11, xfce, chrome as a regular user (not as root) visting https://flathub.org/apps/details/net.davidotek.pupgui2 
2. Clicking install,
3. See the file download,
4.  Right-click the file, Go to Permissions->Allow this file to run as a program, Close
5. Double click the file
6. See error

Appimage
1. Visit https://github.com/DavidoTek/ProtonUp-Qt/releases/
2. Download the AppImage file
3. As above, Allow to run as program
4. Double click the AppImage file
5. 
**Expected behavior**  
The above files should install or run Proton

**Screenshots**  
Flatpak:
![Error](https://user-images.githubusercontent.com/94144702/196236895-840ea1f3-fd3e-415d-a8a8-8e3f51dcba01.png)
Appimage:
![empty appimage](https://user-images.githubusercontent.com/94144702/196238788-9f6644ab-bc2d-4cc5-a7ea-d67a5989db83.png)

**Desktop (please complete the following information):**  
 - OS: Debian 11 Bullseye
 - Version [e.g. ProtonUp-Qt 2.5.0 AppImage or Flatpak] net.davidotek.pupgui2.flatpakref 2.7.4

**Additional context**  
Do not have Steam or Wine etc installed before starting

**Terminal output**  
```
Paste the output of running ProtonUp-Qt from the Terminal here.
AppImage: ./ProtonUp-Qt*.AppImage
gwhiz@ip-172-31-40-67:~/Downloads$ ./ProtonUp-Qt-2.7.4-x86_64.AppImage 
ProtonUp-Qt 2.7.4 by DavidoTek. Build Info: Official AppImage by DavidoTek.
Python 3.8.2 (default, Mar 13 2020, 10:14:16) [GCC 9.3.0], PySide 6.2.4
Platform: Linux-5.10.0-18-cloud-amd64-x86_64-with-glibc2.29
Second instance of ProtonUp-Qt found!
gwhiz@ip-172-31-40-67:~/Downloads$ 

Flatpak: flatpak run net.davidotek.pupgui2
flatpak run net.davidotek.pupgui2
ProtonUp-Qt 2.7.4 by DavidoTek. Build Info: DavidoTek Flathub build.
Python 3.9.9 (main, Nov 10 2011, 15:00:00) [GCC 11.3.0], PySide 6.3.2
Platform: Linux-5.10.0-18-cloud-amd64-x86_64-with-glibc2.33
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vkkhrdisplay, vnc, wayland-egl, wayland, xcb.

/app/bin/net.davidotek.pupgui2: line 3:     3 Aborted                 python3 -m pupgui2 "$@"
gwhiz@ip-172-31-40-67:~/Downloads$ 

```

**Comment by gwhizoftv** [comment]
The above copy/paste of terminal output was from an ssh session. Here is through the terminal in the desktop GUI session:

gwhiz@ip-172-31-40-67:~/.vnc$ cd ../Downloads/
gwhiz@ip-172-31-40-67:~/Downloads$ ./ProtonUp-Qt-2.7.4-x86_64.AppImage 
ProtonUp-Qt 2.7.4 by DavidoTek. Build Info: Official AppImage by DavidoTek.
Python 3.8.2 (default, Mar 13 2020, 10:14:16) [GCC 9.3.0], PySide 6.2.4
Platform: Linux-5.10.0-18-cloud-amd64-x86_64-with-glibc2.29
Second instance of ProtonUp-Qt found!
gwhiz@ip-172-31-40-67:~/Downloads$ flatpak run net.davidotek.pupgui2
ProtonUp-Qt 2.7.4 by DavidoTek. Build Info: DavidoTek Flathub build.
Python 3.9.9 (main, Nov 10 2011, 15:00:00) [GCC 11.3.0], PySide 6.3.2
Platform: Linux-5.10.0-18-cloud-amd64-x86_64-with-glibc2.33
QStandardPaths: wrong permissions on runtime directory /run/user/1001, 0755 instead of 0700
Qt: Session management error: Authentication Rejected, reason : None of the authentication protocols specified are supported and host-based authentication failed
Loaded ctmod GE-Proton
Loaded ctmod Wine-GE
Loaded ctmod Boxtron
Loaded ctmod Kron4ek Wine-Builds Vanilla
Loaded ctmod Lutris-Wine
Loaded ctmod Luxtorpeda
Loaded ctmod Proton Tkg
Loaded ctmod Roberta
cat: /etc/lsb-release: No such file or directory
Loaded ctmod SteamTinkerLaunch
cat: /etc/lsb-release: No such file or directory
Loaded ctmod SteamTinkerLaunch-git
Loaded ctmod DXVK
Loaded ctmod DXVK Async
Loaded ctmod DXVK (nightly)
qt.pysideplugin: Environment variable PYSIDE_DESIGNER_PLUGINS is not set, bailing out.
qt.pysideplugin: No instance of QPyDesignerCustomWidgetCollection was found.
Gamepad error: No gamepad found.

**Comment by gwhizoftv** [comment]
I also tried it using sudo via ssh with these results:

admin@ip-172-31-40-67:~$ sudo flatpak install flathub net.davidotek.pupgui2
Looking for matches…
Required runtime for net.davidotek.pupgui2/x86_64/stable (runtime/org.kde.Platform/x86_64/6.3) found in remote flathub
Do you want to install it? [Y/n]: Y

net.davidotek.pupgui2 permissions:
    ipc      network     wayland     x11     devices     file access [1]     dbus access [2]

    [1] xdg-config/kdeglobals:ro, ~/.bashrc, ~/.config, ~/.kshrc, ~/.local/share/Steam, ~/.local/share/lutris/pga.db:ro, ~/.local/share/lutris/runners,
        ~/.local/share/lutris/runtime/dxvk, ~/.steam, ~/.var/app/com.heroicgameslauncher.hgl/config/heroic/tools,
        ~/.var/app/com.usebottles.bottles/data/bottles/runners, ~/.var/app/com.valvesoftware.Steam/data/Steam, ~/.var/app/net.lutris.Lutris/config/lutris:ro,
        ~/.var/app/net.lutris.Lutris/data/lutris/pga.db:ro, ~/.var/app/net.lutris.Lutris/data/lutris/runners,
        ~/.var/app/net.lutris.Lutris/data/lutris/runtime/dxvk, ~/.zshrc, ~/stl:create
    [2] com.canonical.AppMenu.Registrar, org.freedesktop.Flatpak


        ID                                             Branch            Op           Remote            Download
 1. [✓] org.freedesktop.Platform.GL.default            21.08             i            flathub           130.3 MB / 129.8 MB
 2. [✓] org.freedesktop.Platform.openh264              2.0               i            flathub             1.8 MB / 1.5 MB
 3. [✓] org.kde.KStyle.Adwaita                         6.3               i            flathub             7.8 MB / 7.8 MB
 4. [✓] org.kde.Platform.Locale                        6.3               i            flathub           145.7 MB / 325.8 MB
 5. [✓] org.kde.Platform                               6.3               i            flathub           223.8 MB / 247.8 MB
 6. [✓] net.davidotek.pupgui2                          stable            i            flathub            20.3 MB / 22.3 MB

Warning: Not exporting file net.davidotek.pupgui2.appdata.xml of unsupported type.
Installation complete.

**Comment by DavidoTek** [comment]
Looks fine to me. ProtonUp-Qt will only work (and show entries) when Steam or some other Launcher is installed.

When Steam is installed, it is selected in the box below `Install for:` and you can install a new Proton version by clicking `Add Version`

>  WHen double-clicking Flatpak ref, we get an error "Failed to Install/ File not supported"

Sometimes the `.flatpakref` files do not work properly. If Flatpak and the app store plugin is installed, you can just search for `ProtonUp-Qt` in the app store (e.g. GNOME Software, Discover, ...).
Installing it from Terminal using the command you used (`flatpak install flathub net.davidotek.pupgui2`) works fine too.

> `Second instance of ProtonUp-Qt found!`

Did you run ProtonUp-Qt from Terminal while another window was open? It will only open one window, you have to close the other one. (If there was no other window open, it could also be a bug, try deleting `/tmp/pupgui.a70200`)

**Comment by gwhizoftv** [comment]
@DavidoTek On Debian there is no /etc/lsb-release. Instead lsb-release is a command you run to get the info.
Refer to this: https://github.com/chirpstack/chirpstack-gateway-bridge/issues/30 as an example of an app like ours.

**Comment by DavidoTek** [comment]
> On Debian there is no /etc/lsb-release

That shouldn't matter too much other than for installing SteamTinkerLaunch on a Steam Deck.

I'm wondering, is there a `/etc/os-release` on Debian or where does it get the info?

**Comment by gwhizoftv** [comment]
Yes there is:

***@***.****:*~*$ file !$

file /etc/os-release

/etc/os-release: symbolic link to ../usr/lib/os-release

***@***.****:*~*$ file /usr/lib/os-release

/usr/lib/os-release: ASCII text

***@***.****:*~*$ cat !$

cat /usr/lib/os-release

PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"

NAME="Debian GNU/Linux"

VERSION_ID="11"

VERSION="11 (bullseye)"

VERSION_CODENAME=bullseye

ID=debian

HOME_URL="https://www.debian.org/"

SUPPORT_URL="https://www.debian.org/support"

BUG_REPORT_URL="https://bugs.debian.org/"

***@***.****:*~*$

**Comment by DavidoTek** [comment]
Okay thanks, good to know.

The issue with ProtonUp-Qt should be solved then, I will close it.

**Comment by gwhizoftv** [comment]
> Looks fine to me. ProtonUp-Qt will only work (and show entries) when Steam or some other Launcher is installed.
> 

Is there any writeup that explains why we need a launcher and what it does that Proton does not? I mean can I run a game on Proton without a launcher some how?

**Comment by gwhizoftv** [comment]
@DavidoTek  I installed Lutris but Lutris is not appearing in the launchers pulldown. It is still empty.

**Comment by gwhizoftv** [comment]
I'm sorry you closed it. I've opened a new issue as a result.
